### PR TITLE
Add mw.Echo and the related OOjs and OOUI types

### PR DESCRIPTION
**Pull request by lens0021**

### CODE REVIEW

File: MediaWiki.d.ts [hunk 1]
```diff
@@ -76,6 +132,23 @@ interface MediaWiki {
 	cookie: MwCookie,
 	Map: MwMap,
 	storage: MwStorage,
+	echo?: {
```
Comment (jdlrobson) [inline MediaWiki.d.ts:135]: Could you create an MwEcho type and put these changes in a separate file?

e.g. 

```
echo?: MwEcho
```

I think that would be best as in future if we ever want to change how this works, it should be an easy switch.

**Review by jdlrobson — COMMENTED** [review]
One minor request but this LGTM

File: MediaWiki.d.ts (hunk 1 above)
Comment (lens0021) [reply MediaWiki.d.ts]: Done

**Review by jdlrobson — APPROVED** [review]
Not sure about the long-term implications of including OOUI and Echo definitions (it seems like it would be better to include the typescript definitions in OOUI for example) but these methods seem relatively stable for now.

**Pull request merged**